### PR TITLE
Use function argument in error message instead of return value

### PR DESCRIPTION
**Pull request by jtraglia**
**What type of PR is this?**

Bug fix

**What does this PR do? Why is it needed?**

This fixes 8 instances of incorrect error handling. If a function call returned an error, the (potentially nil) return value was used in the error message. Instead, the error message should include the value passed to the function.

These were identified using Semgrep & the `invalid-usage-of-modified-variable` rule.

**Other notes for review**

References:

* https://semgrep.dev
* https://semgrep.dev/r?q=trailofbits.go.invalid-usage-of-modified-variable.invalid-usage-of-modified-variable
* https://github.com/trailofbits/semgrep-rules/blob/main/go/invalid-usage-of-modified-variable.yml

### CODE REVIEW

File: validator/slashing-protection-history/import.go [hunk 1]
```diff
@@ -341,18 +341,18 @@ func transformSignedAttestations(pubKey [fieldparams.BLSPubkeyLength]byte, atts
...
-			return nil, fmt.Errorf("%d is not a valid epoch: %w", source, err)
+			return nil, fmt.Errorf("%s is not a valid epoch: %w", attestation.SourceEpoch, err)
 		}
 		var signingRoot [32]byte
 		// Signing roots are optional in the standard JSON file.
 		if attestation.SigningRoot != "" {
 			signingRoot, err = RootFromHex(attestation.SigningRoot)
 			if err != nil {
-				return nil, fmt.Errorf("%#x is not a valid root: %w", signingRoot, err)
+				return nil, fmt.Errorf("%s is not a valid root: %w", attestation.SigningRoot, err)
```
[the remark this answers is not in the document]
Comment (prestonvanloon) [reply validator/slashing-protection-history/import.go:355]: nvm it was a string

File: validator/slashing-protection-history/import.go [hunk 2]
```diff
@@ -316,14 +316,14 @@ func transformSignedBlocks(_ context.Context, signedBlocks []*format.SignedBlock
...
-			return nil, fmt.Errorf("%d is not a valid slot: %w", slot, err)
+			return nil, fmt.Errorf("%s is not a valid slot: %w", proposal.Slot, err)
 		}
 		var signingRoot [32]byte
 		// Signing roots are optional in the standard JSON file.
 		if proposal.SigningRoot != "" {
 			signingRoot, err = RootFromHex(proposal.SigningRoot)
 			if err != nil {
-				return nil, fmt.Errorf("%#x is not a valid root: %w", signingRoot, err)
+				return nil, fmt.Errorf("%s is not a valid root: %w", proposal.SigningRoot, err)
```
[the remark this answers is not in the document]
Comment (prestonvanloon) [reply validator/slashing-protection-history/import.go:326]: nvm it was a string already

**Review by prestonvanloon — APPROVED** [review]
LGTM thanks! (Please disregard my prior comments)

**Comment by jtraglia** [comment]
No worries. I'm having trouble understanding why `bazel-build-and-test-everything` failed. Is that normal?

**Comment by prestonvanloon** [comment]
@jtraglia 

```
--- FAIL: Test_uint64FromString (0.00s)
    --- FAIL: Test_uint64FromString/Epoch/Overflow_uint64_gets_MaxUint64 (0.00s)
        helpers_test.go:71: EpochFromString() got = 0, want 18446744073709551615
    --- FAIL: Test_uint64FromString/Slot/Overflow_uint64_gets_MaxUint64 (0.00s)
        helpers_test.go:81: SlotFromString() got = 0, want 18446744073709551615
```

**Comment by jtraglia** [comment]
Ah I see. Lemme fix that. Thanks.

**Comment by jtraglia** [comment]
Fixed. Sorry about that. I thought `ParseUint` would return 0 if there was a problem, it doesn't always.

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**